server/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorSpec.java

```diff
@@ -100,4 +100,16 @@ default Set<ResourceAction> getInputSourceResources() throws UnsupportedOperatio
    * @return source like stream or topic name
    */
   String getSource();
+
+  /**
+   * Checks if a proposed evolution of the supervisor spec is allowed.
+   * <p>
+   * SupervisorSpec `that` is proposed to replace the current supervisor spec. Implementations of this method determine
+   * if the system should allow this evolution.
+   * </p>
+   *
+   * @param that the proposed supervisor spec
+   * @throws IllegalArgumentException if the evolution is not allowed
+   */
+  void validateProposedSpecEvolution(SupervisorSpec that) throws IllegalArgumentException;
```

Comment:
Maybe use a simpler method name like `validateSpecUpdateTo`, and the arg can be called `proposedSpec`.

We should throw a DruidException (of type not supported or invalid input) instead.

Also, please add a default impl to avoid having to override in all impls where this is not needed right now.

Reply:
Thanks for that default tip, cleaned up a ton :) also fixed the Exception type to use DruidException and made an attempt at rename